monkey/infection_monkey/island_api_client/http_client.py

```diff
@@ -60,22 +60,13 @@ def __init__(self, retries=RETRIES):
         retry_config = Retry(retries)
         self._session.mount("https://", HTTPAdapter(max_retries=retry_config))
         self._api_url: Optional[str] = None
+        self._headers = {}
 
-    @handle_island_errors
-    def connect(self, island_server: SocketAddress):
-        try:
-            self._api_url = f"https://{island_server}/api"
-            # Don't use retries here, because we expect to not be able to connect.
-            response = requests.get(  # noqa: DUO123
-                f"{self._api_url}?action=is-up",
-                verify=False,
-                timeout=MEDIUM_REQUEST_TIMEOUT,
-            )
-            response.raise_for_status()
-        except Exception as err:
-            logger.debug(f"Connection to {island_server} failed: {err}")
-            self._api_url = None
-            raise err
+    def set_server(self, server: SocketAddress):
+        self._api_url = f"https://{server}/api"
+
+    def set_authentication_token(self, auth_token: str):
+        self._headers = {"Authentication-Token": auth_token}
```

Comment:
If we leave `self._headers` as an empty dict, you can just `self._headers["Authentication-Token"] = auth_token`, which wouldn't wipe out other headers (not that we have any ATM.)